my-code/src/containers/Home.js

```diff
@@ -0,0 +1,105 @@
+import React, { Component } from 'react'
+import SearchInput from '../components/searchInput';
+import MovieCard from '../components/movieCard';
+import logo from '../assets/images/logo.svg';
+import { withRouter } from 'react-router-dom';
+import EmptyStateManager from '../components/emptyStateManager';
+
+class Home extends Component {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      data: [],
+      searchValue: '',
+      loading: false,
+      error: '',
+      addedMovies: [],
+    }
+  }
+
+  componentDidMount() {
+    if(localStorage.getItem('addedMovies')) {
+      const retrievedData = localStorage.getItem('addedMovies');
+      const parseData = JSON.parse(retrievedData);
+      this.setState({ addedMovies: parseData})
+    }
+  }
+
+  handleChange(event) {
+    const value = event.target.value;
+    this.setState({ searchValue: value,loading: true, error: '' },
+      () => {
+        this.fetchMovies(value);
+      }
+    );
+  }
+
+  fetchMovies(value) {
+    fetch(`http://www.omdbapi.com/?apikey=a6487992&s=${value}` , {
```

Comment:
Never push API keys or secret keys to Github, take a look at dotenv.
It's not a big deal here, as it's a challenge using a public API, but in a real project, it would be a serious issue.

Reply:
I feel ashamed of this rookie mistake...